ipc_sendfile.c

```diff
@@ -0,0 +1,148 @@
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "ipc_common.h"
+
+
+int main(int argc, char *argv[])
+{
+	long int file_size = 0;
+	long int fd;
+	int error_check=0;
+	
+	ipc* info = options(argc, argv);
+
+	if (info->fflag == 1) {
+		fd = open(info->file_name, O_RDONLY,0660);
+		file_size = find_file_size(fd);
+		
+		if (fd == -1) {
+			printf("file name: %s, can not open \n", info->file_name);
+			perror("open");
+			exit(EXIT_FAILURE);
+		}
+		if ( info->pflag + info->qflag + info->sflag != 1) {
+			printf("name of file is used: %s\n", info->file_name);
+			printf("Select one IPC method!e.g \n"
+					"./ipcSend -q /<queue_name> -f <file_name>\n"
+					"./ipcSend -p <pipe_name> -f <file_name>\n"
+					"./ipcSend -s <shared_name> -f <file_name>\n");
+			exit(EXIT_FAILURE);
+
+		} else {
+			if(info->pflag){
+				printf("<pipe is used, pipe name is %s>\n", info->method_name);
+				printf("Send file name: %s\n", info->file_name);
+			} else if (info->qflag) {
+				printf("<queue is used, queue name is %s>\n", info->method_name);
+				printf("Send file name: %s\n", info->file_name);
+				
+				//message queue descriptors
+				mqd_t mq;
+				//mq_unlink(info->method_name);
+
+				// open a message queue using default attributes
+				mq = mq_open(info->method_name, O_CREAT  | O_WRONLY, 0666, NULL);
```

Comment:
Double space (perhaps has no effect though)
```suggestion
				mq = mq_open(info->method_name, O_CREAT | O_WRONLY, 0666, NULL);
```